api/types/resource_153.go

```diff
@@ -348,3 +350,35 @@ func (r *resource153ToUnifiedResourceAdapter) CloneResource() ResourceWithLabels
 	clone := r.inner.(ClonableResource153).CloneResource()
 	return Resource153ToUnifiedResource(clone)
 }
+
+type resource153ToLegacyAdapterV2 struct {
+	resource153ToLegacyAdapter
+}
+
+// MarshalJSON adds support for marshaling the wrapped resource (instead of
+// marshaling the adapter itself).
+// If the resource153 is implemented by a protobuf type, we use
+// protojson.Marshal instead.
+func (r *resource153ToLegacyAdapterV2) MarshalJSON() ([]byte, error) {
```

Comment:
should we add a unmarshalv2?

Reply:
I don't think you can unmarshal into a resource 153 adapter, you unmarshal into the resource directly.

Reply:
yeah, that's correct